frontend/model/state.js

```diff
@@ -92,26 +92,61 @@ sbp('sbp/selectors/register', {
-      const upgradeRequired = Object.entries(state[ourIdentityContractId].groups).forEach(([groupID, { hasLeft }]: [string, Object]) => {
-        if (hasLeft || !state[groupID]?.chatRooms) return
+      Object.entries(state[ourIdentityContractId].groups).map(([groupID, { hasLeft }]: [string, Object]) => {
+        if (hasLeft || !state[groupID]?.chatRooms) return undefined
         // $FlowFixMe[incompatible-use]
-        Object.values((state[groupID].chatRooms: { [string]: Object })).flatMap(({ members }) => {
+        return Object.values((state[groupID].chatRooms: { [string]: Object })).flatMap(({ members }) => {
           return Object.values(members)
-        }).reduce((upgradeRequired: boolean, member: Object) => {
-          return upgradeRequired || (member.status === PROFILE_STATUS.ACTIVE && member.joinedHeight == null)
+        }).reduce((upgradeRequired: string | boolean, member: Object) => {
+          return upgradeRequired || (member.status === PROFILE_STATUS.ACTIVE && member.joinedHeight == null && groupID)
         }, false)
+      }).forEach((contractID) => {
+        if (!contractID) return
+        sbp('gi.actions/group/upgradeFrom1.0.7', { contractID }).catch(e => {
+          console.error('[state/vuex/postUpgradeVerification] Error during gi.actions/group/upgradeFrom1.0.7', contractID, e)
+        })
       })
-      if (!upgradeRequired) return
-      sbp('gi.actions/group/upgradeFrom1.0.7').catch(e => {
-        console.error('[state/vuex/postUpgradeVerification] Error during gi.actions/group/upgradeFrom1.0.7', e)
-      })
+    })();
+    (() => {
+      // Upgrade from version 1.0.8 to a newer version
+      // The new chatroom contracts have an admin IDs list
+      // This code checks if the attribute is missing, and if so, issues the
+      // corresponing upgrade action.
+      const ourIdentityContractId = state.loggedIn?.identityContractID
+      if (!ourIdentityContractId || !state[ourIdentityContractId]) return
+      if (state[ourIdentityContractId].groups) {
+        // Group chatrooms
+        Object.entries(state[ourIdentityContractId].groups).map(([groupID, { hasLeft }]: [string, Object]) => {
+          if (hasLeft || !state[groupID]?.chatRooms || !state[groupID].groupOwnerID) return []
+          // $FlowFixMe[incompatible-use]
+          return Object.entries((state[groupID].chatRooms: { [string]: Object })).flatMap(([chatroomID, { members }]) => {
+            return members[ourIdentityContractId]?.status === PROFILE_STATUS.ACTIVE && !Array.isArray(state[chatroomID]?.attributes?.adminIDs) && [chatroomID, state[groupID].groupOwnerID]
+          })
+        }).forEach(([contractID, groupOwnerID]) => {
+          if (!contractID) return
+          sbp('gi.actions/chatroom/upgradeFrom1.0.8', { contractID, data: groupOwnerID }).catch(e => {
+            console.error('[state/vuex/postUpgradeVerification] Error during gi.actions/chatroom/upgradeFrom1.0.8', contractID, e)
```

Comment:
Shouldn't the chatroom `data` for the `adminIDs` include both the group owner and whoever created the chatroom?

Reply:
Maybe, but that's not how it presently works. Either way, that could probably be done without using adminIDs, but adding a check for the creator.